aws/logs_monitoring/lambda_function.py

```diff
@@ -189,6 +192,12 @@ def enrich(events):
         extract_ddtags_from_message(event)
         extract_host_from_cloudtrails(event)
 
+    for event in reversed(events):
```

Comment:
It seems odd to me for this very Security Hub-specific logic to live in the top-level `enrich()` function. Can we move this logic to the parsing file somehow?

Reply:
Also, why are the events reversed here?

Reply:
I made this suggestion, to simplify the code significantly. Iteration in the reversed order made `remove` and `extend` events  in the loop possible (no side effect).
 
Perhaps @NoisomePossum you can add a comment to explain?

Reply:
Yeah, looping through in reverse order makes it possible to remove each event as we're iterating over it - because you need to replace the original event with a series of events where the data is split between them. Otherwise as you're going forward you end up skipping events when you remove one and then continue iterating.

I don't mind changing this to pass the entire events array into the function to perform the loop in there. However, I was also thinking that there are other logs that have a similar issue with arrays of objects (WAF logs come to mind as one that customers reach out about frequently). In that case, would it be better to keep this reverse loop in case we want to add another function to handle other similar logs? This is actually something I wouldn't mind working on myself so it could end up getting added relatively soon.

Reply:
That makes sense -- I still think there's some value in "hiding" this loop in a function so that it's not right in the main `enrich()` function. Generally when we can abstract details of how things work into lower-level functions, it will make the code of the top-level functions easier to read. It sounds like there might be more than one use for this logic (i.e. not just Security Hub findings), so maybe the lower-level function that reverses the events and separates them could be extended in the future to handle more than one use case.

I also would look into whether this belongs inside the `enrich()` function. The enrich function already does a lot, and the description says "Adds event-specific tags and attributes to each event" -- what's happening here seems like something somewhat different, splitting an event into multiple smaller events. Maybe this logic should happen *after* enrichment, not *during* enrichment.

Reply:
Perhaps we could add an `extract` (or `transform`?) function after the `enrich` like `extract(enrich(...)`, and move these logic to the new function?

Reply:
What do you think @nhinsch ?

Reply:
Went ahead and updated this with a `transform` function.